### PR TITLE
dlt_unit_test: Fix and improve quality of unit tests

### PR DESCRIPTION
**Pull request by minminlittleshrimp**
+ Fix issues offline log unit test fail when not trigger gzip feature
+ Add timeout to prevent timeout silently ignored
+ Limit dlt_user final test suite to 5 seconds
+ Fix wrongly triggered gtests with environment (gateway+offline)

### CODE REVIEW

**Comment by minminlittleshrimp** [comment]
Hello @alexmohr 
For the point: "Limit dlt_user final test suite to 5 seconds",
I would like to have your comment to make sure it is ok to
minimize the max runtime of the test suite.
Link to PR: https://github.com/COVESA/dlt-daemon/pull/362
Thank you

**Comment by minminlittleshrimp** [comment]
### More detail about this fix:

**Running a fully ctest verbose and observed:**

6: Test command: /bin/sh "gtest_dlt_daemon_offline_log.sh" "gtest_dlt_daemon_offline_log"
6: Test timeout computed to be: 9.99988e+06
6/8 Test #6: gtest_dlt_daemon_offline_log ..............   Passed    0.00 sec

**Running gtest_dlt_daemon_offline_log.sh and then ./gtest_dlt_daemon_offline_log
Result:**
 [----------] 3 tests from t_dlt_logstorage_write_on_msg
[ RUN      ] t_dlt_logstorage_write_on_msg.normal
[       OK ] t_dlt_logstorage_write_on_msg.normal (1 ms)
[ RUN      ] t_dlt_logstorage_write_on_msg.gzip
[87723.288506]~DLT~31308~ERROR    ~dlt_logstorage_open_log_file: Unable to open log file.
/home/lum3hc/work/github/minmindlt/backup/dlt-daemon/tests/gtest_dlt_daemon_offline_log.cpp:1212: Failure
Value of: dlt_logstorage_prepare_on_msg(&config, &file_config, path, 1, &newest_file_name)
  Actual: -1
Expected: DLT_RETURN_OK
Which is: 0
Segmentation fault (core dumped)

**Comment by alexmohr** [comment]
> Hello @alexmohr For the point: "Limit dlt_user final test suite to 5 seconds", I would like to have your comment to make sure it is ok to minimize the max runtime of the test suite. Link to PR: #362 Thank you

15s where a bit overly cautious, if there is something broken the test should fail in the 5s you set as timeout, if it doesn't break anything i'd increase to 7s or so just to have a bit more safety margin.

File: tests/gtest_dlt_daemon_offline_log.cpp [hunk 1]
```diff
@@ -647,6 +646,7 @@ TEST(t_dlt_logstorage_get_config, normal)
     num_config = dlt_logstorage_get_config(&handle, config, apid, ctid, ecuid);
 
     EXPECT_EQ(num_config, 3);
+
```
Comment (minminlittleshrimp) [inline tests/gtest_dlt_daemon_offline_log.cpp]: redundant line-> remove

**Comment by minminlittleshrimp** [comment]
The closed PR: https://github.com/COVESA/dlt-daemon/pull/496 is now fixed with new commit version of this PR. To check the change, please kindly try to run $ctest -V and observe gtests running.

**Review by michael-methner — APPROVED** [review]
Looks good to me. Thank you very much.

**Pull request merged**